### DISCUSSION TITLE
Usage in Vue 3 component with script setup

### DISCUSSION THREAD

**Discussion by rico** [permafrost-dev/vue-ray#61] [Q&A] [answered]
Can anybody tell me how to use this in a Vue 3 component with script setup. How can I call the `ray()` method?

**Comment by patinthehat** [comment, marked as answer]
@rico Take a look at the [node-ray](https://github.com/permafrost-dev/node-ray) package, which is used under the hood by this package.

**Reply by rico** [reply]
Aha - thanks, it works. However, _node-ray_ does not seem to offer the `.track(name)` method, which is basically the one I was looking for :-)

**Comment by hrsa** [comment]
@rico how did you manage to make it work?
I've searched around, but didn't understand, if i need to reimport anything in the components.. and any type of this.$ray() doesn't work for me :(

**Comment by rico** [comment]
@hrsa sorry, but I never tried it again - always using the good old `console` and Vue devtools for debugging and insights.

**Comment by patinthehat** [comment]
@rico @hrsa version 2.0 of vue-ray has been released, with script setup support :fire:
